components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -471,9 +516,48 @@ class SystemEngineView @JvmOverloads constructor(
         }
     }
 
+    private fun resetJSAlertAbuseState() {
+        jsAlertCount = 0
+        shouldShowMoreDialogs = true
+    }
+
+    internal fun updateJSDialogAbusedState() {
+        if (!areDialogsWithTimeLimit()) {
+            jsAlertCount = 0
+        }
+        ++jsAlertCount
+        lastDialogShownAt = Date()
+    }
+
+    internal fun areDialogsBeingAbused(): Boolean {
+        return areDialogsWithTimeLimit() || areDialogsWithinCountLimit()
```

Comment:
@Amejia481 yes, that fixed the logic! However, the new names have inverse meaning now:

You're asking `areDialogsBeingAbused` here and that should be the case if dialogs are over (not within) the time limit `or` over (not within) the count limit. Easiest fix: let's keep this new fixed logic and just rename back to `areDialogsAbusedByTime` and `areDialogsAbusedByCount`. Alternatively `areDialogsOverTimeLimit` and `areDialogsOverCountLimit` would also work. So, logic is fine now, but naming needs an update :). Then 🚢 !

Reply:
👍